airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/state/GlobalAsyncStateManager.java

```diff
@@ -175,7 +177,9 @@ public List<PartialStateWithDestinationStats> flushStates() {
 
           // technically possible this map hasn't been updated yet.
           final var oldestStateCounter = stateIdToCounter.get(oldestStateId);
-          Objects.requireNonNull(oldestStateCounter, "Invariant Violation: No record counter found for state message.");
+          if (oldestStateCounter == null) {
```

Comment:
is there a valid situation for this counter being null? (if yes - can you add that as a comment in the code directly?)